src/cargo/util/workspace.rs

```diff
@@ -22,7 +22,7 @@ fn get_available_targets<'a>(
         .map(Target::name)
         .collect();
 
-    targets.sort();
+    targets.sort_unstable();
```

Comment:
This [also came up previously](https://github.com/rust-lang/cargo/pull/8831#discussion_r517433187) and I don't think this change should be made.